File: twoSum.java
public class twoSum {

    public static int[] twoSumFunction(int[] nums, int target) {
        int n = nums.length - 1;
        for (int i = 0; i <= n; i++) {
            for (int j = i + 1; j <= n; j++) {
                if (nums[i] + nums[j] == target) {
                    System.out.println(nums[i]+" "+nums[j]);
                    return new int[] { i, j };
                }
                if (nums[n - i] + nums[j] == target) {
                    System.out.println(n-nums[i]+" "+nums[j]);
                    return new int[] { n - i, j };
                }
            }
        }
        System.out.print("NO SOLUTION");
        return new int[] { -1, -1 };
    }

    public static void main(String[] args) {
        int[] arr = {2,3,4,1,8};
        int target = 8;
        twoSumFunction(arr,target);
    }
}

File: twoSum.js
function twoSum(nums, target) {
  for (let i = 0; i < nums.length; i++) {
    for (let j = i + 1; j < nums.length; j++) {
      if (nums[i] + nums[j] === target) {
        console.log([i, j]);
      }
      if (nums[nums.length - 1 - i] + nums[j] === target) {
        console.log([nums.length - 1 - i, j]);
      }
    }
  }
}
twoSum([2, 7, 11, 15], 9);

function twoSum2(nums, target) {
  const numIndicesMap = {};

  for (let i = 0; i < nums.length; i++) {
    const complement = target - nums[i];

    if (numIndicesMap[complement] !== undefined) {
      // Return the indices of the two numbers
      return [numIndicesMap[complement], i];
    }

    // Store the index of the current number
    numIndicesMap[nums[i]] = i;
  }

  // If no solution is found, return an empty array
  return [];
}

const nums = [2, 7, 11, 15];
const target = 9;
console.log(twoSum2(nums, target)); // Output: [0, 1] (because nums[0] + nums[1] = 2 + 7 = 9)
